Functional test for RHBZ1473203 and some adjustments

ldoktor opened the pull request:
This PR adjusts a bit the "get_console" handling and adds functional test for RHBZ1473203.

apahim commented on tests/avocado/avocado_qemu/test.py:
```diff
@@ -123,7 +123,8 @@ def _handle_prompts(session, username, password, prompt, timeout=10,
     last_chance = False
-
+    # Send enter to refresh output (in case session was attached after boot)
+    session.sendline()
```
ok

apahim commented on tests/avocado/avocado_qemu/test.py:
```diff
@@ -131,9 +131,9 @@ def _handle_prompts(session, username, password, prompt, timeout=10,
-                 r"(?<![Ll]ast )[Ll]ogin:\s*$",  # Don't match "Last Login:"
+                 r"\(or (press|type) Control-D to continue\):\s*",
+                 r"[Gg]ive.*[Ll]ogin:\s*",  # Prompt of rescue mode for SUSE.
+                 r"(?<![Ll]ast )[Ll]ogin:\s*",  # Don't match "Last Login:"
```
Sure. And this can always be improved if we face new corner cases.

apahim commented on tests/avocado/rhbz1473203_numa_hotplug.py:
```diff
@@ -0,0 +1,93 @@
+    Associated bug trackers: RHBZ1473203
+        https://bugzilla.redhat.com/show_bug.cgi?id=1473203
+
+    Fixed in commit kernel commit dc421b200f91930c9c6a9586810ff8c232cf10fc.
```
commit kernel commit

apahim commented on tests/avocado/rhbz1473203_numa_hotplug.py:
```diff
@@ -0,0 +1,93 @@
+        console.close()
+
+    def tearDown(self):
+        self.vm.shutdown()
```
This test requires a vm image. I'd include both the parameters.yaml file and the instructions to use it.

ldoktor commented:
OK, updated. There is yet another batch of fixes to `get_console`, change to default params, reworked way to add image to qemu command and not insignificant changes to the functional test make it less-error-prone.

apahim commented on tests/avocado/test_nec-usb-xhci.py:
```diff
@@ -18,6 +18,7 @@ class TestNecUsbXhci(test.QemuTest):
 
     def setUp(self):
         usbdevice = os.path.join(self.workdir, 'usb.img')
+        self.require_image()
```
`self.request_image()`

ldoktor replied:
sure, sorry I added it on a last minute... I'll fix it in v2

apahim replied:
no need for v2. I'm handling that while integrating your commits.

apahim commented:
All merged.
